src/main/java/org/opensearch/flowframework/workflow/CreateConnectorStep.java

```diff
@@ -0,0 +1,158 @@
+            public void onFailure(Exception e) {
+                logger.error("Failed to create connector");
+                createConnectorFuture.completeExceptionally(new FlowFrameworkException(e.getMessage(), RestStatus.INTERNAL_SERVER_ERROR));
+            }
+        };
+
+        String name = null;
+        String description = null;
+        String version = null;
+        String protocol = null;
+        Map<String, String> parameters = new HashMap<>();
+        Map<String, String> credentials = new HashMap<>();
+        List<ConnectorAction> actions = null;
+
+        for (WorkflowData workflowData : data) {
+            Map<String, Object> content = workflowData.getContent();
+
+            for (Entry<String, Object> entry : content.entrySet()) {
+                switch (entry.getKey()) {
+                    case NAME_FIELD:
+                        name = (String) content.get(NAME_FIELD);
+                        break;
+                    case DESCRIPTION:
+                        description = (String) content.get(DESCRIPTION);
+                        break;
+                    case VERSION_FIELD:
+                        version = (String) content.get(VERSION_FIELD);
+                        break;
+                    case PROTOCOL_FIELD:
+                        protocol = (String) content.get(PROTOCOL_FIELD);
+                        break;
+                    case PARAMETERS_FIELD:
+                        parameters = getParameterMap((Map<String, String>) content.get(PARAMETERS_FIELD));
+                        break;
+                    case CREDENTIALS_FIELD:
+                        credentials = (Map<String, String>) content.get(CREDENTIALS_FIELD);
+                        break;
+                    case ACTIONS_FIELD:
+                        actions = (List<ConnectorAction>) content.get(ACTIONS_FIELD);
+                        break;
+                }
+
+            }
+        }
+
+        if (Stream.of(name, description, version, protocol, parameters, credentials, actions).allMatch(x -> x != null)) {
+            MLCreateConnectorInput mlInput = MLCreateConnectorInput.builder()
+                .name(name)
+                .description(description)
+                .version(version)
+                .protocol(protocol)
+                .parameters(parameters)
+                .credential(credentials)
+                .actions(actions)
+                .build();
+
+            mlClient.createConnector(mlInput, actionListener);
+        }
+
+        return createConnectorFuture;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    private static Map<String, String> getParameterMap(Map<String, String> params) {
+
+        Map<String, String> parameters = new HashMap<>();
+        for (String key : params.keySet()) {
+            String value = params.get(key);
+            try {
+                AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
```

Comment:
Just trying to learn, why this special handling?

Reply:
ml-commons does this special handling for the params map [here](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java#L72. But now I'm thinking do we have to do it on our end as well. Probably not, so I'll go ahead and remove this and just type cast to `Map<String, String>`.